### PR TITLE
Make `FileListResolver` public

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Make the `FileListResolver` public to be able to use it when using ORT programmatically. See the commit messages for details.

### CODE REVIEW

File: scanner/src/test/kotlin/utils/FileListResolverTest.kt [hunk 1]
```diff
@@ -57,7 +57,20 @@ class FileListResolverTest : StringSpec({
...
+            "**/.bzr",
+            "**/.git",
+            "**/.hg",
+            "**/.repo",
+            "**/.svn",
+            "**/CVS",
+            "**/CVSROOT"
+        )
+
+        fileList.files should containExactlyInAnyOrder(
```
Comment (fviernau) [inline scanner/src/test/kotlin/utils/FileListResolverTest.kt]: Is the ordering a feature and should we keep on testing it?

File: scanner/src/test/kotlin/utils/FileListResolverTest.kt [hunk 2]
```diff
@@ -57,7 +57,20 @@ class FileListResolverTest : StringSpec({
 
         val fileList = resolver.resolve(ArtifactProvenance(sourceArtifact = RemoteArtifact.EMPTY))
 
-        fileList.toYaml() shouldBe File("src/test/assets/expected-file-list.yml").readText()
+        fileList.ignorePatterns should containExactlyInAnyOrder(
```
Comment (fviernau) [inline scanner/src/test/kotlin/utils/FileListResolverTest.kt]: Why is independence of serialization valuable here?

File: scanner/src/test/kotlin/utils/FileListResolverTest.kt (hunk 1 above)
Comment (mnonnenmacher) [reply scanner/src/test/kotlin/utils/FileListResolverTest.kt]: The type of `files` is `Set<FileEntry>`, so I would say no.

File: scanner/src/test/kotlin/utils/FileListResolverTest.kt (hunk 2 above)
Comment (mnonnenmacher) [reply scanner/src/test/kotlin/utils/FileListResolverTest.kt]: It makes the test file self-contained and does not require IO for the verification.

File: scanner/src/main/kotlin/utils/FileListResolver.kt [hunk 3]
```diff
@@ -37,10 +37,24 @@ import org.ossreviewtoolkit.utils.common.VCS_DIRECTORIES
 import org.ossreviewtoolkit.utils.common.isSymbolicLink
 import org.ossreviewtoolkit.utils.common.safeDeleteRecursively
 
+/**
+ * A resolver for [FileList]s that are associated with [KnownProvenance]s.
```
[the remark this answers is not in the document]
Comment (mnonnenmacher) [reply scanner/src/main/kotlin/utils/FileListResolver.kt:41]: I'm not sure what exactly you would like to change.
BTW, it's not a cache but a permanent storage.

File: scanner/src/test/kotlin/utils/FileListResolverTest.kt (hunk 2 above)
Comment (fviernau) [reply scanner/src/test/kotlin/utils/FileListResolverTest.kt]: > It makes the test file self-contained and does not require IO for the verification.
hm,
> Why is independence of serialization valuable here? 
...compared to the other cases where we use expected results.

**Review by fviernau — APPROVED** [review]

File: scanner/src/test/kotlin/utils/FileListResolverTest.kt (hunk 2 above)
Comment (mnonnenmacher) [reply scanner/src/test/kotlin/utils/FileListResolverTest.kt]: Yes, but these are often files with hundreds of lines (like for the package managers tests) where inlining would make the test class unreadable. In this case I find it more convenient not to have to look at a separate file.
If you think we should keep testing serialization, especially if the ordering works correctly, I would rather add a separate test case for that (but probably also not use an external file).

Comment (mnonnenmacher) [reply scanner/src/test/kotlin/utils/FileListResolverTest.kt]: See #9540.

Comment (fviernau) [reply scanner/src/test/kotlin/utils/FileListResolverTest.kt]: I've found the separate file to be easier to read, but anyhow - no agreement needed on this.

**Pull request merged**